instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DefaultHttpClientOperation.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.tracer;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+
+public class DefaultHttpClientOperation<RESPONSE> implements HttpClientOperation<RESPONSE> {
+
+  private final Context context;
+  // TODO separate implementation when parentContext is not needed (memory optimization)?
+  private final Context parentContext;
+  // TODO (trask) move end/endExceptionally/etc to here and remove tracer reference
+  private final HttpClientTracer<?, ?, RESPONSE> tracer;
```

Comment:
I'm going to work a bit more on this PR. I'll move it back to draft status for now.

I think it will be useful to see how it looks once we start moving things over here from tracers, e.g.
* do tracers just become Operation factories, exposing a single public method `startOperation`?
* can we strictly limit the public surface of `HttpClientOperation`, so that we don't need a separate no-op implementation for each subclass

Reply:
I took only a cursory look at this and I think I like it. But why "moving things over here from tracers" is a good direction? Why is an `Operation` a better place for a code than a `Tracer`? Because it is not limited to tracing?